tests/test_examples.py

```diff
@@ -1336,30 +1299,84 @@ def test_second_stage_optimization_CoilSet():
     np.testing.assert_allclose(field[0].current, 0, atol=1e-12)
 
 
-# TODO: replace this with the solution to Issue #1021
+@pytest.mark.slow
 @pytest.mark.unit
-def test_optimize_with_fourier_planar_coil():
-    """Test optimizing a FourierPlanarCoil."""
+def test_optimize_with_all_coil_types(DummyCoilSet, DummyMixedCoilSet):
+    """Test optimizing for every type of coil and dummy coil sets."""
+    sym_coils = load(load_from=str(DummyCoilSet["output_path_sym"]), file_format="hdf5")
+    asym_coils = load(
+        load_from=str(DummyCoilSet["output_path_asym"]), file_format="hdf5"
+    )
+    mixed_coils = load(
+        load_from=str(DummyMixedCoilSet["output_path"]), file_format="hdf5"
+    )
+    nested_coils = MixedCoilSet(sym_coils, asym_coils)
```

Comment:
`nested_coils` here actually has a similar structure as `mixed_coils`. What I meant by "nested" coilsets was `MixedCoilSet`s within other `MixedCoilSet`s. So I think we want something like this instead: 
```
nested_coils = MixedCoilSet(mixed_coils, sym_coils, asym_coils, spline_coil)
```
or anything else that contains `mixed_coils` inside another `MixedCoilSet`